src/app/stepprocess/stepprocess-auto-step-change.demo.ts

```diff
@@ -0,0 +1,105 @@
+        ]
+      }, // eslint-disable-line
+      { id: 'step4', title: 'Step 4', icon: 'empty-circle', content: 'Step 4 content' }, // eslint-disable-line
+      {
+        id: 'step5', title: 'Step 5', icon: 'half-empty-circle',
+        substeps: [
+          { id: 'step5Sub1', title: 'Step 5 Sub 1', icon: 'success', content: 'Step 5 Sub 1 content' }, // eslint-disable-line
+          { id: 'step5Sub2', title: 'Step 5 Sub 2', icon: 'empty-circle', content: 'Step 5 Sub 2 content' }, // eslint-disable-line
+          { id: 'step5Sub3', title: 'Step 5 Sub 3', icon: 'error', content: 'Step 5 Sub 3 content' }, // eslint-disable-line
+          { id: 'step5Sub4', title: 'Step 5 Sub 4', icon: 'empty-circle', content: 'Step 5 Sub 4 content' }, // eslint-disable-line
+          { id: 'step5Sub5', title: 'Step 5 Sub 5', icon: 'empty-circle', content: 'Step 5 Sub 5 content' }, // eslint-disable-line
+          { id: 'step5Sub6', title: 'Step 5 Sub 6', icon: 'error', content: 'Step 5 Sub 6 content' }, // eslint-disable-line
+        ]
+      }, // eslint-disable-line
+      { id: 'step6', title: 'Step 6', icon: 'empty-circle', content: 'Step 6 content' }, // eslint-disable-line
+      { id: 'step7', title: 'Step 7', icon: 'error', content: 'Step 7 content' }, // eslint-disable-line
+      { id: 'step8', title: 'Step 8', icon: 'success', content: 'Step 8 content' }, // eslint-disable-line
+      { id: 'step9', title: 'Step 9', icon: 'success', content: 'Step 9 content' }, // eslint-disable-line
+    ];
+  }
+
+  ngAfterViewInit() {
+
+    setTimeout(() => {
+      if (this.stepListItems) {
+        const stepListItemsLength = this.stepListItems.toArray().length;
+        const stepNo = Math.floor(Math.random() * stepListItemsLength);
+        this.stepListItems.toArray()[stepNo].isSelected = true;
+        console.log(stepNo);
+        console.log('step changed');
+      }
+    }, 5000);
+    setTimeout(() => {
+      if (this.stepListItems) {
+        const stepListItemsLength = this.stepListItems.toArray().length;
+        const stepNo = Math.floor(Math.random() * stepListItemsLength);
+        this.stepListItems.toArray()[stepNo].isSelected = true;
+        console.log(stepNo);
+        console.log('step changed');
+      }
+    }, 7000);
+    setTimeout(() => {
+      if (this.stepListItems) {
+        const stepListItemsLength = this.stepListItems.toArray().length;
+        const stepNo = Math.floor(Math.random() * stepListItemsLength);
+        this.stepListItems.toArray()[stepNo].isSelected = true;
+        console.log(stepNo);
+        console.log('step changed');
+      }
+    }, 9000);
+    setTimeout(() => {
+      if (this.stepListItems) {
+        const stepListItemsLength = this.stepListItems.toArray().length;
+        const stepNo = Math.floor(Math.random() * stepListItemsLength);
+        this.stepListItems.toArray()[stepNo].isSelected = true;
+        console.log(stepNo);
+        console.log('step changed');
+      }
+    }, 11000);
+
```

Comment:
lets make it just select one defined random sub step, this way we can add an automated test for it in the future.

Reply:
Actually there's no need for this and the ViewChildren since isSelected is an Input in SohoStepListItemComponent, we can just add `[isSelected]="true"`  to one of the `soho-step-list-item` in the template

Reply:
I just realize http://localhost:4200/ids-enterprise-ng-demo/step-process already does this so maybe we can just use that

Reply:
Hi @nbcp, tried to refactor the code of the demo page, and it will rely on the clicking of the buttons rather than awaiting the setTimeout functions in the ngAfterViewInit.